### PR TITLE
Update: Order Status badge tooltip shows static description  

### PR DESCRIPTION
**Pull request by KIKOmanasijev**
This PR updates the tooltip text over the order status badges in the dashboard. The amount of notes information ("plus 8  other notes" text from the screenshot bellow this paragraph) inside the tooltip is now replaced with static description for the actual order status.

image

Why are we implementing this feature? 
- We noticed that this functionality triggers a noticeable amount of queries (at least X, where X is the amount of posts per page), and there is no way to disable it. In our case, the count of notes and the tooltip itself, was not useful at all, therefore we don't see a reason to have extra queries that slow our site. 
- The company I work at is a huge eCommerce business, that has thousands of orders per day, thus every single millisecond is precious to our team.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes # .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:
 
1. [Current behaviour] Go to WP Dashboard -> WooCommerce -> Orders. Find an order, and hover over it's status. A tooltip should appear, showing the amount of notes that specific order has.
2. [The PR's changes] Go back again to WP Dashboard -> WooCommerce -> Orders. Find an order, and hover it's status. This time a tooltip should appear, but it should show a static description for the orders' status. The status description is derived from [these](https://woocommerce.com/document/managing-orders/order-statuses/) docs.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [X] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [X] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Comment by theroadfreak** [comment]
+1

**Comment by trajche** [comment]
Nice one. This saves us at least 100 queries on our Order page.

**Comment by adrianduffell** [comment]
Thanks for the PR @KIKOmanasijev and surfacing the problem.

I have a few thoughts on this. The tooltip seems to be of little value and not worth generating 100 queries for any site. Arguably we could remove the tooltip altogether because it's not following good UX guidelines:

* Hover tooltips should describe the UI element. It shows the latest note, but I would expect a longer description about the "Completed" label.
* Information should not be hidden in hover tooltips. If notes are important information, they should be displayed on the table itself.
* The tooltip fails when a larger sized note is present. Here is a paragraph of lorem ipsum.
note with lorem ipsum

The notes are also being conflated with status. Users can add any note to provide extra information about the order, it's not intended to be a status update per se.

Rather than introducing a filter, I would recommend changing the tooltip to show a static description of the current order status that we have listed here: https://woocommerce.com/document/managing-orders/order-statuses/


| Label | Tooltip |
|---|---|
| Pending payment | The order has been received, but no payment has been made. Pending payment orders are generally awaiting customer action. |
| On hold | The order is awaiting payment confirmation. Stock is reduced, but you need to confirm payment. |
| Processing | Payment has been received (paid), and the stock has been reduced. The order is awaiting fulfillment. |
| Completed | Order fulfilled and complete. |
| Failed | The customer’s payment failed or was declined, and no payment has been successfully made |
| Draft | Draft orders are created when customers start the checkout process while the block version of the checkout is in place. |
| Canceled | The order was canceled by an admin or the customer. |
| Refunded | Orders are automatically put in the Refunded status when an admin or shop manager has fully refunded the order’s value after payment. |

**Comment by KIKOmanasijev** [comment]
Hey @adrianduffell, thanks for sharing your opinion. I completely agree with what you said, and adjusted my code to implement your suggested changes.

Could you please review it, and let me know your opinion again?

File: plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php
Comment (KIKOmanasijev) [inline plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php:1076]: This filter should be useful for site owners who have custom order statuses.

Comment (adrianduffell) [reply plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php:1076]: Nice idea for the filter! Let's aim for the next available release, so the since tag should be `@since 9.1.0`

Could you also please wrap the strings in the `__()` [translation function](https://codex.wordpress.org/I18n_for_WordPress_Developers)? The lint check will also need single quotes. e.g. 

```
__( 'The order has been received, but no payment has been made. Pending payment orders are generally awaiting customer action.', 'woocommerce' );
```

Comment (KIKOmanasijev) [reply plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php:1076]: Thank you @adrianduffell! 

I've made the requested changes, take a look at them whenever you have the time :)

**Review by adrianduffell — APPROVED** [review]
Thanks for the latest change, @KIKOmanasijev, and apologies not to reply sooner. It's testing well for me and looking rather good!

Let's aim to merge this before the 9.1 code freeze on Jun 19. However, there are a few procedural things to improve before it can be merged.

1. We need a changelog entry. You can add this by following the instructions in the PR description at the bottom.
2. There are a few lint failures listed below. I may be able to push fixes for these to your PR if you prefer.

```
. lint:changes:branch:php: FILE: plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php
. lint:changes:branch:php: -----------------------------------------------------------------------------------------------
. lint:changes:branch:php: FOUND 9 ERRORS AND 0 WARNINGS AFFECTING 9 LINES
. lint:changes:branch:php: -----------------------------------------------------------------------------------------------
. lint:changes:branch:php:  1053 | ERROR | Parameter comment must end with a full stop (Squiz.Commenting.FunctionComment.ParamCommentFullStop)
. lint:changes:branch:php:  1055 | ERROR | Function return type is void, but function contains return statement (Squiz.Commenting.FunctionComment.InvalidReturnVoid)
. lint:changes:branch:php:  1060 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (Universal.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
. lint:changes:branch:php:  1061 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (Universal.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
. lint:changes:branch:php:  1062 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (Universal.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
. lint:changes:branch:php:  1063 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (Universal.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
. lint:changes:branch:php:  1064 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (Universal.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
. lint:changes:branch:php:  1065 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (Universal.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
. lint:changes:branch:php:  1066 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (Universal.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
. lint:changes:branch:php: -----------------------------------------------------------------------------------------------
.
```

**Comment by KIKOmanasijev** [comment]
> Thanks for the latest change, @KIKOmanasijev, and apologies not to reply sooner. It's testing well for me and looking rather good!
> 
> Let's aim to merge this before the 9.1 code freeze on Jun 19. However, there are a few procedural things to improve before it can be merged.
> 
> 1. We need a changelog entry. You can add this by following the instructions in the PR description at the bottom.
> 2. There are a few lint failures listed below. I may be able to push fixes for these to your PR if you prefer.
> 
> ```
> . lint:changes:branch:php: FILE: plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php
> . lint:changes:branch:php: -----------------------------------------------------------------------------------------------
> . lint:changes:branch:php: FOUND 9 ERRORS AND 0 WARNINGS AFFECTING 9 LINES
> . lint:changes:branch:php: -----------------------------------------------------------------------------------------------
> . lint:changes:branch:php:  1053 | ERROR | Parameter comment must end with a full stop (Squiz.Commenting.FunctionComment.ParamCommentFullStop)
> . lint:changes:branch:php:  1055 | ERROR | Function return type is void, but function contains return statement (Squiz.Commenting.FunctionComment.InvalidReturnVoid)
> . lint:changes:branch:php:  1060 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (Universal.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
> . lint:changes:branch:php:  1061 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (Universal.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
> . lint:changes:branch:php:  1062 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (Universal.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
> . lint:changes:branch:php:  1063 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (Universal.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
> . lint:changes:branch:php:  1064 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (Universal.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
> . lint:changes:branch:php:  1065 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (Universal.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
> . lint:changes:branch:php:  1066 | ERROR | Spaces must be used for mid-line alignment; tabs are not allowed (Universal.WhiteSpace.DisallowInlineTabs.NonIndentTabsUsed)
> . lint:changes:branch:php: -----------------------------------------------------------------------------------------------
> .
> ```

I just uploaded the changelog, but the Linter is throwing some errors on my end, could you please run it for me? Thanks 🙇‍♂️

**Comment by adrianduffell** [comment]
Hey @KIKOmanasijev, I've updated the lint issues and also resetted some of the indentation that was made elsewhere in the file. One final request, are you able to please update the PR title and description now that the change is different from the original?

**Comment by KIKOmanasijev** [comment]
Thank you @adrianduffell. I just updated the PR title, and the description.

**Comment by WilkoMarketing** [comment]
Hi team,

We found the detailed information previously shown in the tooltips very useful for our operations. Do you have any plans to reintroduce that detailed note information in the tooltips or offer an alternative way to access it? Your consideration would be greatly appreciated.

Thank you!

**Comment by KIKOmanasijev** [comment]
Hey @WilkoMarketing, in my opinion, I should have used the $order as an argument inside the "woocommerce_get_order_status_labels" filter, which would have been helpful in your case. 

Take a look:
https://github.com/KIKOmanasijev/woocommerce/blob/c5b3c8d32db059175d22fb455e6a7e886b98bdd6/plugins/woocommerce/src/Internal/Admin/Orders/ListTable.php#L1076

image

**Comment by naman03malhotra** [comment]
Hey folks, can you please take a look at issues: https://github.com/woocommerce/woocommerce/issues/50284 and https://github.com/woocommerce/woocommerce/issues/50434 seems like some folks have issue with the new UX.

**Comment by lynnjat7** [comment]
8573330-zd-a8c another report / vote for the old way. I understand from the change made in 9.1 that this is intended and we have no plans to revert to the old way. Is there the possibility of making this an option?

**Comment by ilyasfoo** [comment]
Hey, @lynnjat7 and @WilkoMarketing!

Currently there's no intention to revert, but in WooCommerce 9.2 we'll have an option to use a code snippet to restore the old functionality. I'll keep you posted!

**Comment by ilyasfoo** [comment]
Hi @lynnjat7 and @WilkoMarketing!

You can use [this snippet](https://gist.github.com/ilyasfoo/2d614e01b2100f2b476e235bd4cdba06) to restore the old tooltip for WooCommerce 9.2.0 or greater. Here's how to download it as a plugin:

1. Open the link https://gist.github.com/ilyasfoo/2d614e01b2100f2b476e235bd4cdba06
2. Click on `Download ZIP` button
3. Change the zip file name if you prefer and and click on save
4. Go to your WooCommerce site, upload and activate the plugin
5. Go to `WooCommerce > Orders` and observe the tooltips are restored

**Comment by JanisE** [comment]
Wow, removed a feature just like that! Because two people were not using it... Not just an option to disable it but removed altogether...

I agree that the status tooltip may not be the correct place for the latest order note, but for our case, the latest note info was very useful on the order list page, as we are actually actively utilizing them, additional notes written by our shop administrators manually and by related systems via API.

As it is now, you've just replaced a very useful (for some) information with some almost useless static text (as all the shop administrators have long memorized what each of the status means).

In order to not just remove a functionality, a better way would be to move the info from the tool-tip to a separate column on the order list, like so:

```php
<?php

class woo_admin_columns {
  private static $instance;

  private function __construct() {
    $this->init();
  }

  public static function get_instance() {
    if (!isset(self::$instance)) {
      self::$instance = new self;
    }

    return self::$instance;
  }

  public function hide_custom_columns_by_default($hidden, $screen){
    if (isset($screen->id) && $screen->id == 'edit-shop_order') {
      $hidden = array_merge($hidden, ['order_last_note',]);
    }

    return $hidden;
  }

  public function add_list_columns($in_columns) {
    $columns = [];

    // Inserting columns to a specific location
    foreach($in_columns as $key => $column){
      $columns[$key] = $column;

      if($key == 'order_status'){
        // Inserting after "Status" column.
        $columns['order_last_note'] = __('Last note', 'woocommerce');
      }
    }

    return $columns;
  }

  protected static function get_order_latest_note($order_id){
    $latest_note_html = '';

    $latest_notes = wc_get_order_notes([
      'order_id' => $order_id,
      'limit'    => 1,
      'orderby'  => 'date_created_gmt',
    ]);

    $latest_note = current($latest_notes);
    $approved_comments_count = absint(get_comment_count($order_id)['approved']);

    if(isset($latest_note->content)){
      $latest_note_html = $latest_note->content;

      if($approved_comments_count > 1){
        /* translators: %d: notes count */
        $latest_note_html .= '<br/><small style="display:block">' . sprintf(_n('Plus %d other note', 'Plus %d other notes', ($approved_comments_count - 1), 'woocommerce'), $approved_comments_count - 1) . '</small>';
      }
    }
    else{
      /* translators: %d: notes count */
      $latest_note_html = sprintf(_n('%d note', '%d notes', $approved_comments_count, 'woocommerce'), $approved_comments_count);
    }

    return wp_kses_post($latest_note_html);
  }

  public function add_custom_column_data($column, $order_id) {
    switch($column){
      case 'order_last_note':
        echo self::get_order_latest_note($order_id);
        break;
    }
  }

  private function init() {
    if(is_admin()){
      add_action('init', function() {load_plugin_textdomain('woocommerce');});

      add_filter('default_hidden_columns', [$this, 'hide_custom_columns_by_default'], 10, 2);
      add_filter('manage_edit-shop_order_columns', [$this, 'add_list_columns'], 20);
      add_action('manage_shop_order_posts_custom_column', [$this, 'add_custom_column_data'], 20, 2);
    }
  }
}
```

If hidden columns are still contributing to the server load, that's another issue altogether, that should be solved in another way.

**Comment by trajche** [comment]
I just want to point out that this merge has improved our admin performance *significantly* dropping those queries. 

Nice workaround by @JanisE for those that need it. Perhaps adding it to 'Screen Options' is another fix so that you can enable/disable that column there.

**Pull request merged**
